engine/common/follower/engine.go

```diff
@@ -141,18 +141,18 @@ func New(
 }
 
 // OnBlockProposal errors when called since follower engine doesn't support direct ingestion via internal method.
```

Comment:
```suggestion
// OnBlockProposal logs an error and drops the proposal. This is because the follower ingests new
// blocks directly from the networking layer (channel `channels.ReceiveBlocks` by default), which
// delivers its messages by calling the generic `Process` method. Receiving block proposal as 
// from another internal component is likely an implementation bug.
```